article/_posts/2019-10-23-Humain-Ai-Coord.md

```diff
@@ -0,0 +1,89 @@
+    title:   "On the Utility of Learning about Humans for Human-AI Coordination"
+    venue:   "NeurIPS 2019"
+pdf: ""
+---
+
+
+# Highlights
+- Self-play algorithms use (near) optimal solutions when playing together against humans
+- However, RL algorithms do not perform as well when playing _with_ humans
+- Training them to perform with humans makes them perform better with humans
+
+# Introduction
+
+> An increasingly effective way to tackle two-player games is to train an agent to play with a set of other AI agents, often past versions of itself. This powerful approach has resulted in impressive performance against human experts in games like Go, Quake, Dota, and Starcraft.
+
+However, the agents never encounter humans during training and therefore undergo serious distributional shift. How do they not fail catastrophically when put against human players ? The authors hypothesize that it is because of the _competitive_ nature of these games. Indeed, even with a simple _Minimax_<sup>[1](#1)</sup> policy, if a human makes a suboptimal move, the AI agent can only beat them even more soundly.
+
+However, one could argue that the ultimate goal of AI is to produce _collaborative_ agents rather than competitive ones. Recent results in games like Dota<sup>[2](#2)</sup> and Capture the Flag<sup>[2](#3)</sup>. While the results of these experiments seems to point towards self-play extending nicely to cooperation, the authors argue that the nice results come from the agents' own abilities and not coordination with humans.
+
+# Methods
+
+The authors propsed three testable hypotheses and devised a new environment to test them. The hypotheses go as follows:
+
+**H1. A self-play agent will perform much more poorly when partnered with a human (relative to being partnered with itself).**
+
+**H2. When partnered with a human, a human-aware agent will achieve higher performance than a self-play agent, though not as high as a self-play agent partnered with itself.**
+
+**H3. When partnered with a human, a human-aware agent will achieve higher performance than an agent trained via imitation learning.**
+
+![](/article/images/human-ai-coord/gamedyns.png)
+![](/article/images/human-ai-coord/env.jpeg)
+
+They also created the _Overcooked_ environment, based on the video game of the same name, to measure human-agent cooperation. The environment has five different possible scenarios, as pictured above, all with the same goal: Cooking onion soup and delivering it to the serving location.
+
+> Agents should learn how to navigate the map, interact with objects, drop the objects off in the right locations, and finally serve completed dishes to the serving area. All the while, agents should be aware of what their partner is doing and coordinate with them effectively.
+
+To train the agents, the authors gathered 16 human-human trajectories for each layout. They then used behavior cloning to create the $$BC$$ model (training time) and $$H_{proxy}$$ model (test time).
```

Comment:
@jhlegarreta 
> Also, if they use BC, where does the real interaction/playing with humans take place? Are the human-human trajectories an improvement proposed by the authors ?

Before the actual training takes place, as stated here. Obviously, having actual humans in the training loop would be infeasible, hence using behavior cloning to mimic humans.

Reply:
OK.